hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/regression/system/LifecycleTest.java

```diff
@@ -147,7 +202,247 @@ default HapiSpecOperation prepareFakeUpgrade() {
+                sourcing(() -> freezeUpgrade()
+                        .startingIn(2)
+                        .seconds()
+                        .withUpdateFile(DEFAULT_UPGRADE_FILE_ID)
+                        .havingHash(upgradeFileHashAt(FAKE_UPGRADE_ZIP_LOC))),
+                confirmFreezeAndShutdown(),
+                blockingOrder(preRestartOps),
+                doingContextual(spec -> setCurrentConfigVersion(spec, version)));
+    }
+
+    /**
+     * Returns an operation that starts the selected nodes with the current configuration version.
+     *
+     * @param selector the selector for the nodes to start
+     * @param envOverrides the environment overrides to use
+     * @return the operation that starts the nodes
+     */
+    default SpecOperation resumeUpgradedNetwork(
+            @NonNull final NodeSelector selector, @NonNull final Map<String, String> envOverrides) {
+        requireNonNull(selector);
+        requireNonNull(envOverrides);
+        return sourcingContextual(spec -> FakeNmt.startNodes(selector, currentConfigVersion(spec), envOverrides));
+    }
+
+    /**
+     * Returns an operation that starts the selected nodes with the current configuration version.
+     *
+     * @param selector the selector for the nodes to start
+     * @return the operation that starts the nodes
+     */
+    default SpecOperation resumeUpgradedNetwork(@NonNull final NodeSelector selector) {
+        return resumeUpgradedNetwork(selector, Map.of());
+    }
+
+    /**
+     * Returns an operation that starts the selected nodes with the current configuration version
+     * without refreshing override-network.json files.
+     *
+     * @param selector the selector for the nodes to start
+     * @param envOverrides the environment overrides to use
+     * @return the operation that starts the nodes
+     */
+    default SpecOperation resumeUpgradedNetworkWithoutOverrides(
+            @NonNull final NodeSelector selector, @NonNull final Map<String, String> envOverrides) {
+        requireNonNull(selector);
+        requireNonNull(envOverrides);
+        return sourcingContextual(
+                spec -> FakeNmt.startNodesNoOverride(selector, currentConfigVersion(spec), envOverrides));
+    }
+
+    /**
+     * Returns an operation that starts the selected nodes with the current configuration version
+     * without refreshing override-network.json files.
+     *
+     * @param selector the selector for the nodes to start
+     * @return the operation that starts the nodes
+     */
+    default SpecOperation resumeUpgradedNetworkWithoutOverrides(@NonNull final NodeSelector selector) {
+        return resumeUpgradedNetworkWithoutOverrides(selector, Map.of());
+    }
+
+    /**
+     * Returns an operation that upgrades the network to the next configuration version and defers
+     * starting the given node ids until after existing nodes are ACTIVE and a signed state is copied.
```

Comment:
Why are we deferring only _some_ of the nodes, while waiting for others to become active? For network topology changes?